src/uu/mv/src/mv.rs

```diff
@@ -751,3 +845,370 @@ fn is_empty_dir(path: &Path) -> bool {
+            let mut file2_path = sub_dir_path.clone();
+            file2_path.push("test2.txt");
+            let content2 = "content2";
+            fs_extra::file::write_all(&file2_path, content2).unwrap();
+            assert!(file2_path.exists());
+
+            let mut exist_path = path_to.clone();
+            exist_path.push(test_name);
+            create(&exist_path, true).unwrap();
+            assert!(exist_path.exists());
+            exist_path.push(same_file);
+            let exist_content = "exist content";
+            assert_ne!(exist_content, content1);
+            fs_extra::file::write_all(&exist_path, exist_content).unwrap();
+            assert!(exist_path.exists());
+
+            let dir_size = get_size(&path_from).expect("failed to get dir size");
+            let pb = if with_progress_bar {
+                Some(
+                    ProgressBar::new(dir_size).with_style(
+                        ProgressStyle::with_template(
+                            "{msg}: [{elapsed_precise}] {wide_bar} {bytes:>7}/{total_bytes:7}",
+                        )
+                        .unwrap(),
+                    ),
+                )
+            } else {
+                None
+            };
+            move_dir(&path_from, &path_to, pb.as_ref(), None).unwrap();
+            assert!(exist_path.exists());
+            assert!(path_to.exists());
+            assert!(!path_from.exists());
+            if let Some(pb) = pb {
+                assert_eq!(pb.position(), dir_size);
+            }
+        }
+    }
+
+    #[test]
+    fn it_move_inside_work_target_dir_not_exist() {
+        for with_progress_bar in [false, true] {
+            let temp_dir = tempdir().unwrap();
+            let path_root = PathBuf::from(temp_dir.path());
+            let root = path_root.join("it_move_inside_work_target_dir_not_exist");
+            let root_dir1 = root.join("dir1");
+            let root_dir1_sub = root_dir1.join("sub");
+            let root_dir2 = root.join("dir2");
+            let file1 = root_dir1.join("file1.txt");
+            let file2 = root_dir1_sub.join("file2.txt");
+
+            create_all(&root_dir1_sub, true).unwrap();
+            fs_extra::file::write_all(&file1, "content1").unwrap();
+            fs_extra::file::write_all(&file2, "content2").unwrap();
+
+            if root_dir2.exists() {
+                remove(&root_dir2).unwrap();
+            }
+
+            assert!(root_dir1.exists());
+            assert!(root_dir1_sub.exists());
+            assert!(!root_dir2.exists());
+            assert!(file1.exists());
+            assert!(file2.exists());
+            let dir_size = get_size(&root_dir1).expect("failed to get dir size");
+            let pb = if with_progress_bar {
+                Some(
+                    ProgressBar::new(dir_size).with_style(
+                        ProgressStyle::with_template(
+                            "{msg}: [{elapsed_precise}] {wide_bar} {bytes:>7}/{total_bytes:7}",
```

Comment:
Are there any tests that ensure that the values actually make sense? It doesn't have to be all of them, but I don't see any at all that assert specific value(range)s for bytes, total_bytes, etc.

Maybe I missed something?

Reply:
i'm sorry, could you explain more about this